// Protocols used for sending data to Producer Node.
syntax = "proto3";
package airbloc.producer;

message Identifier {
    string type = 1;
    string identifier = 2;
}

message RawData {
    string categoryOfApp = 1;
    Identifier ownerIdentifier = 2;
    string payload = 3;
}

message AddDataResult {
    bool succeed = 1;
    string dataId = 2;
}

service Producer {
    rpc AddData(stream RawData) returns (stream AddDataResult) {}
}